target/linux/d1/patches-6.6/0009-ASoC-sunxi-sun4i-codec-add-basic-support-for-D1-audi.patch

```diff
@@ -16,75 +16,7 @@ Signed-off-by: Maksim Kiselev <bigunclemax@gmail.com>
 
 --- a/sound/soc/sunxi/sun4i-codec.c
 +++ b/sound/soc/sunxi/sun4i-codec.c
-@@ -232,15 +232,65 @@
- 
- /* TODO H3 DAP (Digital Audio Processing) bits */
- 
-+/*
-+ * sun20i D1 and similar codecs specific registers
-+ *
-+ * Almost all registers moved on D1, including ADC digital controls,
-+ * FIFO and RX data registers. Only DAC control are at the same offset.
-+ */
-+
-+#define SUN20I_D1_CODEC_DAC_VOL_CTRL		(0x04)
-+#define SUN20I_D1_CODEC_DAC_VOL_SEL			(16)
-+#define SUN20I_D1_CODEC_DAC_VOL_L			(8)
-+#define SUN20I_D1_CODEC_DAC_VOL_R			(0)
-+#define SUN20I_D1_CODEC_DAC_FIFOC		(0x10)
-+#define SUN20I_D1_CODEC_ADC_FIFOC		(0x30)
-+#define SUN20I_D1_CODEC_ADC_FIFOC_EN_AD			(28)
-+#define SUN20I_D1_CODEC_ADC_FIFOC_RX_SAMPLE_BITS	(16)
-+#define SUN20I_D1_CODEC_ADC_FIFOC_RX_TRIG_LEVEL		(4)
-+#define SUN20I_D1_CODEC_ADC_FIFOC_ADC_DRQ_EN		(3)
-+#define SUN20I_D1_CODEC_ADC_VOL_CTRL1		(0x34)
-+#define SUN20I_D1_CODEC_ADC_VOL_CTRL1_ADC3_VOL		(16)
-+#define SUN20I_D1_CODEC_ADC_VOL_CTRL1_ADC2_VOL		(8)
-+#define SUN20I_D1_CODEC_ADC_VOL_CTRL1_ADC1_VOL		(0)
-+#define SUN20I_D1_CODEC_ADC_RXDATA		(0x40)
-+#define SUN20I_D1_CODEC_ADC_DIG_CTRL		(0x50)
-+#define SUN20I_D1_CODEC_ADC_DIG_CTRL_ADC3_CH_EN		(2)
-+#define SUN20I_D1_CODEC_ADC_DIG_CTRL_ADC2_CH_EN		(1)
-+#define SUN20I_D1_CODEC_ADC_DIG_CTRL_ADC1_CH_EN		(0)
-+#define SUN20I_D1_CODEC_VRA1SPEEDUP_DOWN_CTRL	(0x54)
-+
-+/* TODO D1 DAP (Digital Audio Processing) bits */
-+
-+struct sun4i_codec;
-+
-+struct sun4i_codec_quirks {
-+	const struct regmap_config *regmap_config;
-+	const struct snd_soc_component_driver *codec;
-+	struct snd_soc_card * (*create_card)(struct device *dev);
-+	struct reg_field reg_dac_fifoc;	/* used for regmap_field */
-+	struct reg_field reg_adc_fifoc;	/* used for regmap_field */
-+	unsigned int adc_drq_en;
-+	unsigned int rx_sample_bits;
-+	unsigned int rx_trig_level;
-+	unsigned int reg_dac_txdata;	/* TX FIFO offset for DMA config */
-+	unsigned int reg_adc_rxdata;	/* RX FIFO offset for DMA config */
-+	bool has_reset;
-+	bool has_dual_clock;
-+};
-+
- struct sun4i_codec {
- 	struct device	*dev;
- 	struct regmap	*regmap;
- 	struct clk	*clk_apb;
--	struct clk	*clk_module;
-+	struct clk	*clk_module; /* used for ADC if clocks are separate */
-+	struct clk	*clk_module_dac;
- 	struct reset_control *rst;
- 	struct gpio_desc *gpio_pa;
-+	const struct sun4i_codec_quirks *quirks;
- 
--	/* ADC_FIFOC register is at different offset on different SoCs */
-+	/* DAC/ADC FIFOC registers are at different offset on different SoCs */
-+	struct regmap_field *reg_dac_fifoc;
- 	struct regmap_field *reg_adc_fifoc;
- 
- 	struct snd_dmaengine_dai_dma_data	capture_dma_data;
-@@ -250,33 +300,33 @@ struct sun4i_codec {
+@@ -252,33 +252,33 @@ struct sun4i_codec {
```

Comment:
I think these changes are still needed. The changes between 6.6.92 and 6.6.93 are looking different: https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/diff/sound/soc/sunxi/sun4i-codec.c?id=v6.6.93&id2=v6.6.92

Reply:
Thank you for your review and the correct observation. This is fixed.